### PR TITLE
[WP8] Respect SplashScreen and SplashScreenDelay preferences from config file

### PR DESCRIPTION
**Pull request by galexandrov**
Extend WP8 Splash Screen to respect SplashScreen and SplashScreenDelay preferences from config file

### CODE REVIEW

File: src/wp/SplashScreen.cs [hunk 1]
```diff
@@ -68,27 +77,52 @@ public override void OnInit()
             if (!WasShown)
             {
                 WasShown = true;
-                show();
+                this.show();
             }
         }
 
-        void LoadConfigValues()
+        private static void GetPreference(XDocument document, string preferenceName, Action<string> action)
```
Comment (robpaveza) [inline src/wp/SplashScreen.cs]: This method works somewhat backwards, where you check up to N strings for every preference, making your worst-case performance O(n^2) (in the case that you don't find the preferences).  Suggested alternative: Populate a Dictionary<string, Action<string>>, and then for-each preference element, if (dictionary.TryGetValue(preferenceName, out action)) action(value).  Dictionary lookups approach O(1).

File: src/wp/SplashScreen.cs [hunk 2]
```diff
@@ -40,26 +40,35 @@ namespace WPCordovaClassLib.Cordova.Commands
...
+            if (imageResource != null)
+            {
+                BitmapImage splash_image = new BitmapImage();
+                splash_image.SetSource(imageResource.Stream);
+                SplashScreen.Source = splash_image;
+            }
 
             // Instansiate the popup and set the Child property of Popup to SplashScreen
-            popup = new Popup() {IsOpen = false, Child = SplashScreen };
+            this.popup = new Popup() { IsOpen = false, Child = SplashScreen };
```
Comment (robpaveza) [inline src/wp/SplashScreen.cs]: Nit: prefixed use of "this" when referring to members isn't bad, but is unnecessary.

File: src/wp/SplashScreen.cs [hunk 3]
```diff
@@ -163,5 +190,30 @@ public void hide(string options = null)
                 story.Begin();
             });
         }
+
+        private void StartAutoHideTimer()
+        {
+            var timer = new DispatcherTimer() { Interval = TimeSpan.FromMilliseconds(preferences.SplashScreenDelay) };
+            timer.Tick += (object sender, EventArgs e) =>
```
Comment (robpaveza) [inline src/wp/SplashScreen.cs]: Suggest: Promote the Tick handler to a member function.  In the callback, remove the Tick handler.  This eliminates the circular reference which is created by auto-referencing 'this' in your callback.

File: src/wp/SplashScreen.cs [hunk 4]
```diff
@@ -163,5 +190,30 @@ public void hide(string options = null)
...
+            {
+                this.hide();
+                timer.Stop();
+            };
+            timer.Start();
+        }
+
+        private class Preferences
```
Comment (robpaveza) [inline src/wp/SplashScreen.cs]: It's not clear that another heap-allocated object is worth it here.  With only three preferences, in a private class, each of these could be promoted to a member variable of the owning class, eliminating the private class altogether.

File: src/wp/SplashScreen.cs (hunk 1 above)
Comment (robpaveza) [inline src/wp/SplashScreen.cs]: Variable name 'document' should probably be 'configuration' or 'configFile'.

File: src/wp/SplashScreen.cs [hunk 5]
```diff
@@ -40,26 +40,35 @@ namespace WPCordovaClassLib.Cordova.Commands
     /// </summary>
     public class SplashScreen : BaseCommand
     {
+        private readonly Preferences preferences = new Preferences();
         private Popup popup;
-        private bool autohide = true;
 
         private static bool WasShown = false;
```
Comment (robpaveza) [inline src/wp/SplashScreen.cs]: Nits: Why static?  Why PascalCased for a private?

**Comment by purplecabbage** [comment]
Thanks for the contribution.
I did some cleanup in general of this code, some per comments from @robpaveza 
( 6b1e72c6312f1c449d15a11be445b858c6689d6a )
The code has been merged.

**Pull request merged**